source/bao_hyp/config.rst

```diff
@@ -3,7 +3,78 @@ Bao Configuration File
 
 Overview
 --------
-An overview of the configuration file options (structure overview)
+The Bao hypervisor's configuration is managed through a dedicated configuration
+in the form of a C source file. This section provides an in-depth
+description of the various configuration options available.
+
+The configuration file follows a specific structure, as outlined below:
+
+.. code-block:: c
+
+    #include <config.h>
+
+    // Load guests' image
+    VM_IMAGE(img1_name, "/path/to/vm1/binary.bin");
+    VM_IMAGE(img2_name, "/path/to/vm2/binary.bin");
+
+    struct config config = {
+
+        // Shared memories configuration
+        .shmemlist_size = N,
+        .shmemlist = (struct shmem[]) {
+            [0] = {/*shared memory config*/,},
+            [1] = {/*shared memory config*/,},
+            ...
+            [N] = {/*shared memory config*/,}
+        },
+
+        // Guests Configuration
+        .vmlist_size = NUM_VMs,
+        .vmlist = {
+            { /* VM 0 Config*/},
+            { /* VM 1 Config*/},
+            ...
+            { /* VM N Config*/},
+        }
+    };
+
+Before defining the configuration struct, it's crucial to declare the VM
+images. The configuration file anticipates receiving a compiled image of each
+guest (.bin file) which will then be allocated to individual VMs using the
+``VM_IMAGE`` macro. This macro allows for the direct embedding of the guest
+binary file into the hypervisor binary file compiled program. Here's an example
+usage of the ``VM_IMAGE``:
+
+.. code-block:: c
+
+  VM_IMAGE(img_name, "/path/to/VM/binary.bin");
+
+In this context, ``img_name`` is the identifier associated with the image that
+will later be used to describe the image running on the VM - `Guest Image`_.
+
+After loading the images, the configuration allows the definition of two
+distinct lists: (i) a list of shared memory regions (``shmemlist``) and (ii) a
+list of VMs (or guests) (``vmlist``). While the list of shared memory regions
+is optional and may be omitted from the configuration, the list of VMs is
+mandatory and must include at least 1 VM. Additionally, for each list, it is
+necessary to specify the list size using the parameters ``shmemlist_size`` and
+``vmlist_size``.
+
+Config file name and location
```

Comment:
I think this section is not clear enough and needs to go deeper in the options available.

This are the missing ideas:
    - the build system looks for the configurations in a specific folder which we call "configuration repo";
    - by default the configuration repo is the *configs* folder in the hypervisors top level directory;
    - the configuration repo can be changed by passing a diffent folder using the ``CONFIG_REPO`` option in the make command when building the hypervisor;
    - Considering a config named `config-name` configs can be palced in two forms:
        1) directly a C source file in the form of *config-name.c*
        2) a directory *config-name* with a the C source file *config.c* inside it.

As @danielRep mention in another comment, it would be useful to briefly mention how the *config-name* will be used when building the hypervisor.

Reply:
Addressed in commit #a02d816